armory/scenarios/poison.py

```diff
@@ -398,85 +478,41 @@ def finalize_results(self):
+        )
+        if self.test_set_class_labels != self.train_set_class_labels:
             log.warning(
                 "Test set contains a strict subset of train set classes.  Some metrics for missing classes may not be computed."
             )
-
-        if self.use_filtering_defense:
-
-            removed = (1 - self.indices_to_keep).astype(np.bool)
-            poisoned = np.zeros_like(self.y_clean).astype(np.bool)
-            poisoned[self.poison_index.astype(np.int64)] = True
-
-            false_negatives = int(np.sum(~removed & poisoned))
-            true_positives = int(np.sum(removed & poisoned))
-            true_negatives = int(np.sum(~removed & ~poisoned))
-            false_positives = int(np.sum(removed & ~poisoned))
-
-            false_negative_rate = 0 if n_poisoned == 0 else false_negatives / n_poisoned
-            true_positive_rate = 0 if n_poisoned == 0 else true_positives / n_poisoned
-            true_negative_rate = true_negatives / n_clean
-            false_positive_rate = false_positives / n_clean
-
-            f1_score = true_positives / (
-                true_positives + 0.5 * (false_positives + false_negatives)
-            )
-
-            self.results["filter_true_positives"] = true_positives
-            self.results["filter_false_positives"] = false_positives
-            self.results["filter_true_negatives"] = true_negatives
-            self.results["filter_false_negatives"] = false_negatives
-            self.results["filter_true_positive_rate"] = true_positive_rate
-            self.results["filter_false_positive_rate"] = false_positive_rate
-            self.results["filter_true_negative_rate"] = true_negative_rate
-            self.results["filter_false_negative_rate"] = false_negative_rate
-            self.results["filter_f1_score"] = f1_score
-            self.results["filter_fraction_data_removed"] = removed.mean()
-            self.results["filter_N_samples_removed"] = int(removed.sum())
-
-            for y in train_set_class_labels:
-                self.results[f"class_{y}_N_train_samples_removed"] = int(
-                    np.sum(self.y_clean[removed] == y)
-                )
-
-        for y in train_set_class_labels:
+        for y in self.train_set_class_labels:
             self.results[f"class_{y}_N_train_samples"] = int(np.sum(self.y_clean == y))
-        for y in test_set_class_labels:
+        for y in self.test_set_class_labels:
             self.results[f"class_{y}_unpoisoned_test_accuracy"] = np.mean(
                 self.benign_test_accuracy_per_class[y]
             )
 
-        if hasattr(self, "fairness_metrics") and not self.check_run:
-            # Get unpoisoned test set
-            dataset_config = self.config["dataset"]
-            test_dataset = config_loading.load_dataset(
-                dataset_config, split="test", num_batches=None, **self.dataset_kwargs
-            )
+    def finalize_results(self):
```

Comment:
Does the poisoning scenario not use MetricsLogger at all?

Reply:
It has not done so as long as I've been around, nor does it use the metrics section of the config.  I suppose because some poisoning metrics don't fit into the mold assumed by the metricslogger class?  I haven't looked closely.  But it will need an overhaul with the measurement pr anyway